cg/meta/workflow/analysis.py

```diff
@@ -0,0 +1,341 @@
+            primary=(len(case_obj.analyses) == 0),
+        )
+        new_analysis.family = case_obj
+        self.status_db.add_commit(new_analysis)
+        LOG.info(f"Analysis successfully stored in StatusDB: {case_id} : {analysis_start}")
+
+    def get_deliverables_file_path(self, case_id: str) -> Path:
+        raise NotImplementedError
+
+    def get_analysis_finish_path(self, case_id: str) -> Path:
+        raise NotImplementedError
+
+    def add_pending_trailblazer_analysis(self, case_id) -> None:
+        if self.trailblazer_api.is_latest_analysis_ongoing(case_id=case_id):
+            LOG.error("Analysis still ongoing in Trailblazer!")
+            raise CgError
+        self.trailblazer_api.mark_analyses_deleted(case_id=case_id)
+        self.trailblazer_api.add_pending_analysis(
+            case_id=case_id,
+            email=environ_email(),
+            type=self.get_application_type(self.status_db.family(case_id).links[0].sample),
+            out_dir=self.get_trailblazer_config_path(case_id=case_id).parent.as_posix(),
+            config_path=self.get_trailblazer_config_path(case_id=case_id).as_posix(),
+            priority=self.get_priority_for_case(case_id=case_id),
+            data_analysis=str(self.pipeline),
+        )
+
+    def get_hermes_transformed_deliverables(self, case_id: str) -> dict:
+        return self.hermes_api.create_housekeeper_bundle(
+            bundle_name=case_id,
+            deliverables=self.get_deliverables_file_path(case_id=case_id),
+            pipeline=str(self.pipeline),
+            analysis_type=self.get_bundle_deliverables_type(case_id),
+            created=self.get_bundle_created_date(case_id),
+        ).dict()
+
+    def get_bundle_created_date(self, case_id: str) -> dt.datetime:
+        raise NotImplementedError
+
+    def get_pipeline_version(self, case_id: str) -> str:
+        raise NotImplementedError
+
+    def set_statusdb_action(self, case_id: str, action: Optional[str]) -> None:
+        """
+        Set one of the allowed actions on a case in StatusDB.
+        """
+        if action in [None, *CASE_ACTIONS]:
+            case_obj: models.Family = self.status_db.family(case_id)
+            case_obj.action = action
+            self.status_db.commit()
+            LOG.info("Action %s set for case %s", action, case_id)
+            return
+        LOG.warning(
+            f"Action '{action}' not permitted by StatusDB and will not be set for case {case_id}"
+        )
+
+    def get_analyses_to_clean(self, before: dt.datetime) -> List[models.Analysis]:
+        analyses_to_clean = self.status_db.analyses_to_clean(pipeline=self.pipeline, before=before)
+        return analyses_to_clean.all()
+
+    def get_cases_to_analyze(self) -> List[models.Family]:
+        return self.status_db.cases_to_analyze(
+            pipeline=self.pipeline, threshold=self.threshold_reads
+        )
+
+    def get_running_cases(self) -> List[models.Family]:
```

Comment:
I definitely think that apis should not talk to the database directly. Suggest to move this to the Store module. Wether it is a part of the API or just a function in a file does not matter to much

Reply:
done